monkey/monkey_island/cc/server_setup.py

```diff
@@ -13,41 +15,71 @@
 from monkey_island.cc.resources.monkey_download import MonkeyDownload  # noqa: E402
 from monkey_island.cc.server_utils.bootloader_server import BootloaderHttpServer  # noqa: E402
 from monkey_island.cc.server_utils.consts import MONKEY_ISLAND_ABS_PATH  # noqa: E402
 from monkey_island.cc.server_utils.encryptor import initialize_encryptor  # noqa: E402
+from monkey_island.cc.server_utils.island_logger import reset_logger, setup_logging  # noqa: E402
 from monkey_island.cc.services.initialize import initialize_services  # noqa: E402
 from monkey_island.cc.services.reporting.exporter_init import populate_exporter_list  # noqa: E402
 from monkey_island.cc.services.utils.network_utils import local_ip_addresses  # noqa: E402
 from monkey_island.cc.setup.mongo.database_initializer import init_collections  # noqa: E402
-from monkey_island.cc.setup.mongo.mongo_setup import MONGO_URL, setup_mongodb  # noqa: E402
+from monkey_island.cc.setup.mongo.mongo_setup import (  # noqa: E402
+    MONGO_URL,
+    connect_to_mongodb,
+    register_mongo_shutdown_callback,
+    start_mongodb,
+)
 from monkey_island.setup.island_config_options import IslandConfigOptions  # noqa: E402
 
 logger = logging.getLogger(__name__)
 
 
-def setup_island(setup_only: bool, config_options: IslandConfigOptions, server_config_path: str):
-    env_singleton.initialize_from_file(server_config_path)
+def run_monkey_island():
+    island_args = parse_cli_args()
+    config_options, server_config_path = _setup_data_dir(island_args)
 
-    initialize_encryptor(config_options.data_dir)
-    initialize_services(config_options.data_dir)
+    _configure_logging(config_options)
+    _initialize_global_resources(config_options, server_config_path)
 
-    bootloader_server_thread = Thread(
-        target=BootloaderHttpServer(MONGO_URL).serve_forever, daemon=True
-    )
+    if config_options.start_mongodb:
+        mongo_db_process = start_mongodb(config_options.data_dir)
+        register_mongo_shutdown_callback(mongo_db_process)
 
-    bootloader_server_thread.start()
-    _start_island_server(setup_only, config_options)
-    bootloader_server_thread.join()
+    connect_to_mongodb()
 
+    _start_island_server(island_args.setup_only, config_options)
 
-def _start_island_server(should_setup_only, config_options: IslandConfigOptions):
 
-    setup_mongodb(config_options)
+def _setup_data_dir(island_args: IslandCmdArgs) -> Tuple[IslandConfigOptions, str]:
+    try:
+        return config_setup.setup_data_dir(island_args)
+    except OSError as ex:
+        print(f"Error opening server config file: {ex}")
+        exit(1)
+    except json.JSONDecodeError as ex:
+        print(f"Error loading server config: {ex}")
+        exit(1)
+
+
+def _configure_logging(config_options):
+    reset_logger()
+    setup_logging(config_options.data_dir, config_options.log_level)
+
+
+def _initialize_global_resources(config_options: IslandConfigOptions, server_config_path: str):
```

Comment:
Resources in the island's codebase mean request endpoints, because it's a REST application. Maybe rename to global_assets or global_services or just globals?

Reply:
I'll go with `_initialize_globals()`.